src/Dispatcher/GroupPosBased.php

```diff
@@ -22,16 +22,16 @@ protected function dispatchVariableRoute(array $routeData, string $uri): array
             for ($i = 1; $matches[$i] === ''; ++$i) {
             }
 
-            [$handler, $varNames] = $data['routeMap'][$i];
+            [$handler, $varNames, $route] = $data['routeMap'][$i];
 
             $vars = [];
             foreach ($varNames as $varName) {
                 $vars[$varName] = $matches[$i++];
             }
 
-            return [self::FOUND, $handler, $vars];
+            return Result::fromArray([self::FOUND, $handler, $vars, $route]);
         }
 
-        return [self::NOT_FOUND];
+        return Result::createNotFound();
```

Comment:
Instantiating an object here will impact performance since this method is called many times during dispatching.